### PR TITLE
Additional-Unit Test

### PR DESCRIPTION
**Pull request by EmilyKatdell**
image

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate

### CODE REVIEW

**Comment by santhoshatdell** [comment]
Workflow check for formatting failed. Please check.

File: controllers/replication-controller/dellcsireplicationgroup_controller.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ type ReplicationGroupReconciler struct {
 	EventRecorder      record.EventRecorder
 	PVCRequeueInterval time.Duration
 	Config             connection.MultiClusterClient
+	ReplicationClient  *csireplication.Replication
```
Comment (santhoshatdell) [inline controllers/replication-controller/dellcsireplicationgroup_controller.go:59]: May I know why is this added here?

File: controllers/replication-controller/dellcsireplicationgroup_controller_test.go [hunk 2]
```diff
@@ -424,6 +426,21 @@ func (suite *RGControllerTestSuite) TestRGSyncDeletion() {
 	suite.Equal(false, resp.Requeue)
 }
 
+func (suite *RGControllerTestSuite) TestNoDeletionTimeStamp() {
```
Comment (santhoshatdell) [inline controllers/replication-controller/dellcsireplicationgroup_controller_test.go:429]: Is this testing a successful reconcile? In that case, can you rename the function name. For ex: TestRGSync?
It simply checks for no error at the end. Could you check for some annotations that was set on RG?

File: controllers/replication-controller/persistentvolume_controller_test.go [hunk 3]
```diff
@@ -473,9 +473,19 @@ func TestPVReconcileSuite(t *testing.T) {
 	suite.Run(t, testSuite)
 }
 
+func (suite *PVReconcileSuite) TestPVNotFound() {
+	req := suite.getTypicalRequest("non-existent-pv")
+	_, err := suite.reconciler.Reconcile(context.Background(), req)
+	suite.NoError(err, "No error on PV reconcile")
```
Comment (santhoshatdell) [inline controllers/replication-controller/persistentvolume_controller_test.go:479]: Test name and assertion do not match. There will be an error returned if PV is not found.

**Review by santhoshatdell — CHANGES_REQUESTED** [review]

File: controllers/replication-controller/dellcsireplicationgroup_controller.go (hunk 1 above)
Comment (lukeatdell) [reply controllers/replication-controller/dellcsireplicationgroup_controller.go:59]: +1. We shouldn't need to add the replication client to this controller for a unit test.

Comment (EmilyKatdell) [reply controllers/replication-controller/dellcsireplicationgroup_controller.go:59]: I have updated that

File: controllers/replication-controller/dellcsireplicationgroup_controller_test.go [hunk 4]
```diff
@@ -424,6 +425,36 @@ func (suite *RGControllerTestSuite) TestRGSyncDeletion() {
...
+	}
+    suite.T().Log(rg.Finalizers, "RG finalizer added")
+    // Check if remote RG got created
+    rClient, err := suite.config.GetConnection(suite.driver.RemoteClusterID)
+    suite.NoError(err)
+    _, err = rClient.GetReplicationGroup(context.Background(), rg.Name)
+    suite.NoError(err)
+    
+    //Delete rg
+    err = suite.client.Delete(context.Background(), rg)
```
Comment (santhoshatdell) [inline controllers/replication-controller/dellcsireplicationgroup_controller_test.go:453]: Could you explain what this test is for? When RG is deleted, there will be deletionTimeStamp set but the test name says 'noDeletionTimeStamp'.

File: controllers/replication-controller/persistentvolume_controller_test.go [hunk 5]
```diff
@@ -473,9 +473,19 @@ func TestPVReconcileSuite(t *testing.T) {
 	suite.Run(t, testSuite)
 }
 
+func (suite *PVReconcileSuite) TestPVNotFound() {
+	req := suite.getTypicalRequest("non-existent-pv")
+	_, err := suite.reconciler.Reconcile(context.Background(), req)
+	suite.NoError(err, "No on PV reconcile")
```
Comment (santhoshatdell) [inline controllers/replication-controller/persistentvolume_controller_test.go:479]: If no PV is found, then there should be an error returned. In that case, we should be checking for suite.Error().

**Comment by santhoshatdell** [comment]
Please update the PR description with the usual format (description, GitHub issue, etc).
